### PR TITLE
#144-Get target info

### PR DESCRIPTION
**Pull request by amandhamija98**
Gets the following information about a selected target contact:
1. name
2. last seen
3. phone
4. about
5. groups in common

### CODE REVIEW

File: wplay/utils/target_search.py [hunk 1]
```diff
@@ -56,6 +61,93 @@
...
+            whatsapp_selectors_dict['contact_info_page_group_element_heading'],
+            visible= True,
+            timeout=3000
+        )
+
+        assert (await page.evaluate(f'document.querySelector("{whatsapp_selectors_dict["contact_info_page_group_element_heading"]}").innerText'))\
+               == "Groups in common"
+        group_elements = await page.querySelectorAll(whatsapp_selectors_dict['contact_info_page_group_elements'])
+        complete_target_info['Groups'] = [await ele.querySelectorEval('div>div>div:nth-child(2)>div:first-child>div>div>span', 'e => e.getAttribute("title")') for ele in group_elements]
+    except Exception as e:
```
Comment (rpotter12) [inline wplay/utils/target_search.py]: Codacy Issue found: [Unused variable 'e'](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5138744)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5138744&bid=16883305)

File: wplay/utils/target_search.py [hunk 2]
```diff
@@ -56,6 +61,93 @@
...
+
+async def get_contact_groups_common_with_target(complete_target_info,page):
+    try:
+        await page.waitForSelector(
+            whatsapp_selectors_dict['contact_info_page_group_element_heading'],
+            visible= True,
+            timeout=3000
+        )
+
+        assert (await page.evaluate(f'document.querySelector("{whatsapp_selectors_dict["contact_info_page_group_element_heading"]}").innerText'))\
```
Comment (rpotter12) [inline wplay/utils/target_search.py]: Codacy Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5138744)

File: wplay/utils/target_search.py (hunk 1 above)
Comment (amandhamija98) [reply wplay/utils/target_search.py]: Sure, will remove that

File: wplay/utils/target_search.py (hunk 2 above)
Comment (amandhamija98) [reply wplay/utils/target_search.py]: I will use an If condition instead

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5138744&bid=16883305)

File: wplay/utils/target_search.py [hunk 3]
```diff
@@ -56,6 +61,95 @@
...
+    except Exception as e:
+        print(e)
+    return contact_page_elements
+
+
+async def get_contact_name_info(contact_name_element,complete_target_info):
+    try:
+        complete_target_info['Name'] = await contact_name_element.querySelectorEval('span > span', 'element => element.innerText')
+        complete_target_info['Last_seen'] = await contact_name_element.querySelectorEval('div > span:last-of-type > div > span', 'element => element.getAttribute("title")')
+    except Exception as e:
```
Comment (rpotter12) [inline wplay/utils/target_search.py]: Codacy Issue found: [Unused variable 'e'](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5138744)

**Comment by xandao6** [comment]
How this works? I couldn't make it work

**Comment by amandhamija98** [comment]
You will have to run it with -m switch
python -m wplay -wchat <name>

**Comment by amandhamija98** [comment]
If you are running it like that and still getting an error could you please send a screenshot

**Comment by xandao6** [comment]
I know, but it should work with all scripts (wchat, wblast, wtrack ...) ?

**Comment by amandhamija98** [comment]
Aah ok, let me check

**Comment by amandhamija98** [comment]
I've run the project for all scripts, it's working fine and giving all info about a target contact. Could you please send screenshots of the error you're facing

**Comment by xandao6** [comment]
Nice work

**Pull request merged**